src/main/src/main/java/org/geoserver/catalog/impl/CascadedLayerInfoImpl.java

```diff
@@ -0,0 +1,254 @@
+                    .getStore()
+                    .getWebMapServer(null)
+                    .getCapabilities()
+                    .getRequest()
+                    .getGetMap()
+                    .getFormats()
+                    .stream()
+                    .filter(CascadedLayerInfoImpl::isImage)
+                    . // only image formats
+                    collect(Collectors.toList());
+        } catch (IOException e) {
+            LOGGER.log(
+                    Level.SEVERE,
+                    "Unable to fetch available formats for cascaded layer " + layerInfo.getName());
+            LOGGER.log(Level.SEVERE, e.getMessage());
+        }
+        return Collections.EMPTY_LIST;
+    }
+
+    @Override
+    public String getPrefferedFormat() {
+        return this.prefferedFormat;
+    }
+
+    @Override
+    public void setPrefferedFormat(String prefferedFormat) {
+        this.prefferedFormat = prefferedFormat;
+    }
+
+    @Override
+    public boolean isSelectedRemoteFormat(String format) {
+        if (prefferedFormat.equalsIgnoreCase(format)) return true;
+        else return selectedRemoteFormats.contains(format);
+    }
+
+    public static boolean isImage(String format) {
+        return format.startsWith("image") && !format.contains("xml") && !format.contains("svg");
+    }
+
+    @Override
+    public Set<StyleInfo> getStyles() {
+        // legacy functionality for empty
+        if (getRemoteStyles().isEmpty()) return super.getStyles();
+
+        return getRemoteStyleInfos()
+                .stream()
+                .filter(s -> !forcedRemoteStyle.equalsIgnoreCase(s.getName()))
+                .filter(s -> selectedRemoteStyles.contains(s.getName()))
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public StyleInfo getDefaultStyle() {
+        if (forcedRemoteStyle != null)
+            if (!forcedRemoteStyle.isEmpty()) {
+                Optional<StyleInfo> defaultRemoteStyle =
+                        getRemoteStyleInfos()
+                                .stream()
+                                .filter(s -> s.getName().equalsIgnoreCase(forcedRemoteStyle))
+                                .findFirst();
+
+                if (defaultRemoteStyle.isPresent()) return defaultRemoteStyle.get();
+            }
+        // default to legacy if layer has no forcedRemoteStyle(unlikely)
+        return super.getDefaultStyle();
+    }
+
+    @Override
+    public Set<StyleInfo> getRemoteStyleInfos() {
+        WMSLayerInfo layerInfo = (WMSLayerInfo) super.resource;
+
+        try {
+            return layerInfo
+                    .getWMSLayer(null)
+                    .getStyles()
+                    .stream()
+                    .map(CascadedLayerInfoImpl::getStyleInfo)
+                    .collect(Collectors.toSet());
+
+        } catch (IOException e) {
+            LOGGER.severe(e.getMessage());
+        }
+        // on error default to super
+        return super.getStyles();
```

Comment:
Why?

Reply:
keeping native functionality intact, I am open to better altenate

Reply:
what do you mean by native functionality? In term of the user workflow I configure or I'm configuring  a remote layer, if I cannot obtain the remote styles then it shoudl crash no? and tell the user looks something bad happen when trying to reach the remote server and get the available styles.

Reply:
understood and agreed 👍